coderd/experiments_test.go

```diff
@@ -0,0 +1,103 @@
+package coderd_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/coderd/httpmw"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/testutil"
+)
+
+// nolint:paralleltest  // t.Setenv
+func Test_Experiments(t *testing.T) {
+	// In these tests we set the environment variable directly, rather than
+	// modifying the deployment config. This corresponds more directly to what
+	// an end user would do. Also, modifying the deployment config doesn't work
+	// for expanding the wildcard because of how viper binds stuff.
+	t.Run("empty", func(t *testing.T) {
+		client := coderdtest.New(t, nil)
+		_ = coderdtest.CreateFirstUser(t, client)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		experiments, err := client.Experiments(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, experiments)
+		require.Empty(t, experiments)
+		require.False(t, experiments.Enabled(codersdk.ExperimentVSCodeLocal))
+		require.False(t, experiments.Enabled("foo"))
+	})
+
+	t.Run("multiple features", func(t *testing.T) {
+		t.Setenv("CODER_EXPERIMENTAL", "foo,bar")
```

Comment:
Instead of using Setenv (which requires no Parallel), try using `coderdtest.DeploymentConfig(t)` and updating the value on that.

Reply:
That might work if I move it out of `DeploymentConfig`. It doesn't work for the current incarnation because the munging happens when `coderdtest.DeploymentConfig(t)` is called.

Reply:
Done!